PwaUriHandler/explainer.md

```diff
@@ -0,0 +1,250 @@
+
+### Manifest Changes
+
+We propose adding a new _optional_ data field `url_handlers` to the manifest object. This data value will contain an array of URI handler declaration objects. Each object should contains a `base` string, an optional `paths` array of strings, and an optional `excludePaths` array of strings.
+
+The `url_handlers` data serves as requests from the PWA to handle URIs. The browser should validate that the PWA has the authority to handle those URIs and then store the request for later use. On an OS that allows for deeper integration, the browser should also perform the URI handling registrations with the OS and keep them in sync with the app lifecycle.
+
+Example web app manifest hosted at `www.contoso.com`:
+
+```json
+{
+  "name": "Contoso App",
+  "description": "A Business App",
+  "uri_handlers": [
+      {
+          "base": "www.contoso.com",
+          "paths": ["*"],
+          "excludePaths": [
+              "/about",
+              "/blog",
+              "/privacy"
+          ]
+      },
+      {
+          "base": "%*.contoso.com",
+          "paths": ["*"]
+      },
+      {
+          "base": "www.conto.so",
+      }
+  ],
+  "icons": [
+    {
+      "src": "images/icons-144.png",
+      "type": "image/png",
+      "sizes": "144x144"
+    }
+  ],
+  "background_color": "#FFFFFF",
+  "display": "standalone",
+  "start_url": "/",
+}
+```
+
+These are the fields in each URI handler object:
+
+| Field     | Required / Optional | Description                                      | Default                           |
+|:----------|:--------------------|:-------------------------------------------------|:----------------------------------|
+| `base`    | Required            | Base portion of the URI to be handled. Can also be a complete URI. | N/A             |
+| `paths`   | Optional            | Array of allowed paths relative to `base`        | `[]`                              |
+| `excludePaths` | Optional       | Array of disallowed paths relative to `base`     | `[]`                              |
+
+A URI matches a URI handler if it matches the `base`, at least one of values in `paths` if there are any, and does not match any of the values in `excludePaths`.
+
+Requested URIs do not have to be within the requesting PWA's scope. In this scheme, any URI can be registered as part of the URI handling request. The `base` field is necessary because URIs from different domains can be requested. Not restricting URIs to the same scope or domain as the requesting PWA gives the developer freedom to use multiple domain names for the same content and handle them with the same PWA. See [this section](#pwa-to-site-association) for how cross-domain requests are validated. The `base` field can start with a `%*.` prefix to indicate the inclusion of subdomains.
```

Comment:
Consider stating, `In this scheme, any verifiably-associated URL` or similar. That way you're leading this paragraph with acknowledgement that PWAs can't register as handlers for any arbitrary URL.

Reply:
How about: 

"In this scheme, a URI matches a URI handler if it matches the `base`, at least one of values in `paths` if there are any, does not match any of the values in `excludePaths`, and is [verifiably associated](#pwa-to-site-association)."

Reply:
That works! Maybe "verifiably associated with the PWA"

Reply:
Edited.